logger.js

```diff
@@ -58,10 +62,17 @@ function pinoLogger (opts, stream) {
     res.startTime = Date.now()
     if (!req.res) { req.res = res }
 
-    res.on('finish', onResFinished)
-    res.on('error', onResFinished)
-    // it's possible that browser aborts connection, or http-server because of timeout
-    req.on('aborted', onReqAborted)
+    res.once('finish', onResFinished)
+    res.once('error', onResFinished)
+
+    req.once('close', onReqClosed)
+    req.once('aborted', onReqAborted)
+    req.onSocketTimeout = function () {
+      req.removeListener('aborted', onReqAborted)
+      res.statusCode = 408
+      onResFinished.call(res, new Error('Timeout'))
+    }
+    req.socket.once('timeout', req.onSocketTimeout)
```

Comment:
@mcollina: We could listen to `res.on('timeout', ...)` as well. The problem is that it will actually change the behavior of the associated socket.